### PR TITLE
Fixes

### LINKED ISSUE #2065: Builder not acknowledging build requests
### Minecolonies version

Happens in at least Minecolonies versions 0.8.6471 and up for 1.12, as well as 0.8.6960 for 1.11.2 ( I didn't check earlier versions than those, so more versions might be affected as well). 

### Expected behavior

- Builder builds buildings

### Actual behaviour

- After finishing the builders' hut and the town hall, the builder just stands around and ignores any further build requests. 
- Placing another builders' hut and hiring a second builder has similar results - the new builder will build their own builders hut and one more building before ignoring further requests. 
- Firing/killing the builder and hiring a new one does not affect anything, the new one will continue to ignore all build requests
- The build requests appear in the work order list in the town hall, but no existing builder is assigned to them
- Cancelling build requests and adding new ones does nothing
- Builder appears able to make decorations prior to the second building, but not after (i.e. the builder can make a builders hut, several fancy paths, and then a town hall - but once the town hall is built no more decorations can be made)

### Steps to reproduce the problem

1. Start a new colony 
2. Build two buildings
3. Builder forgets how to build
4. Profit

I'm not sure how useful it is, but [here's a log](https://pastebin.com/YzJADL2g).

### LINKED ISSUE #2084: Oh come ON!!! Yet again my DMan is broken!!!
### Minecolonies version

version 1.12.2-0.8.7133

### Expected behaviour

- The game is playable!

### Actual behaviour

- Yet again my deliveryman has stopped working.

### Steps to reproduce the problem

1. Install Minecolonies
2. Regularly contribute on Patreon
3. Get fed up with all my colonies getting fatally broken
4. I may stop playing ;-(

Yes, OK, I am upset and not necessarily being rational, but my colonies keep ending up either totally broken or with a delivery man who will no longer work.

In this case I has built a Taiga warehouse (which is not anything like as good as the wooden one in usability in my opinion, even though it looks nice). I upgraded the warehouse to level 3 and had 2 builders busy in the colony serviced by a single deliveryman. Then the dman stopped working with the warehouse. I tried firing/re-employing, employing a different colonist, getting a builder to repair the warehouse, manually breaking some racks then using the builder to repair again (so new racks). Still no good. So, I demolished the warehouse and got the builder to build a new one (wood this time). And the deliveryman is still stick "dumping".

I have attached the colonies.dat and colony1.dat files in case someone knows how to de-kaput them!!

[colonies.dat.zip](https://github.com/Minecolonies/minecolonies/files/1659903/colonies.dat.zip)

### LINKED ISSUE #2085: [1.10.2] Server crash on Tick
### Minecolonies version

version 1.10.2-0.9.7104.jar

**1.**

PM
Description: Exception in server tick loop

java.lang.NullPointerException: Exception in server tick loop
	at com.minecolonies.coremod.colony.CitizenData.setWorkBuilding(CitizenData.java:610)
	at com.minecolonies.coremod.colony.CitizenData.onRemoveBuilding(CitizenData.java:534)
	at com.minecolonies.coremod.colony.managers.BuildingManager.removeBuilding(BuildingManager.java:363)
	at com.minecolonies.coremod.colony.buildings.AbstractBuilding.destroy(AbstractBuilding.java:749)
	at java.util.ArrayList.forEach(Unknown Source)
	at com.minecolonies.coremod.colony.managers.BuildingManager.cleanUpBuildings(BuildingManager.java:204)
	at com.minecolonies.coremod.colony.Colony.onWorldTick(Colony.java:610)
	at com.minecolonies.coremod.colony.ColonyManager.lambda$onWorldTick$1(ColonyManager.java:695)
	at java.util.ArrayList.forEach(Unknown Source)
	at com.minecolonies.coremod.colony.ColonyManager.onWorldTick(ColonyManager.java:695)
	at com.minecolonies.coremod.event.FMLEventHandler.onWorldTick(FMLEventHandler.java:55)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_888_FMLEventHandler_onWorldTick_WorldTickEvent.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185)
	at net.minecraftforge.fml.common.FMLCommonHandler.onPreWorldTick(FMLCommonHandler.java:282)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:694)
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:387)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:613)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471)
	at java.lang.Thread.run(Unknown Source)

**2.** 
also on 1.10 there is still this issue on server startup:

Time: 12/22/17 9:05 PM
Description: Exception in server tick loop

java.lang.NullPointerException: Exception in server tick loop
at exterminatorjeff.undergroundbiomes.common.block.UBOreSedimentary.func_180642_a(UBOreSedimentary.java:42)
at com.minecolonies.api.compatibility.CompatabilityManager.discoverOres(CompatabilityManager.java:162)
at com.minecolonies.api.compatibility.CompatabilityManager.discover(CompatabilityManager.java:63)
at com.minecolonies.coremod.colony.ColonyManager.readFromNBT(ColonyManager.java:818)
at com.minecolonies.coremod.colony.ColonyManager.onWorldLoad(ColonyManager.java:716)
at com.minecolonies.coremod.event.EventHandler.onWorldLoad(EventHandler.java:397)
at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_814_EventHandler_onWorldLoad_Load.invoke(.dynamic)
at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185)
at net.minecraft.server.MinecraftServer.func_71247_a(MinecraftServer.java:263)
at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:266)
at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:431)
at java.lang.Thread.run(Unknown Source)

### LINKED ISSUE #2086: Babysitting the Builder
### Minecolonies version

minecolonies-universal-1.12.2-0.8.7133

### Expected behavior

- Builder checks chest and finds items for build

### Actual behaviour

- Have to pull from chest and give it to him directly

### Steps to reproduce the problem

1. Recalled him
2. exited and restarted world
3. picked up items and put them back, didn't take them and came back, only filling request works.
4. Profit

### LINKED ISSUE #2087: Builder Breaking
1.11.2-0.8.4904

Upgrading my Builder's Hut

Saying "I'm waiting for the rain to stop" when there was no rain.
1. Have some rain appear
2. do /weather clear to clear the weather
3. sometimes the builder will forever say "I'm waiting for the rain to stop
4. Killing them spawns a new person, but they will forever be looking for a job.

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #2087
Closes #2086
Closes #2085
Closes #2084
Closes #2065

# Changes proposed in this pull request:
- Fixes several issues with the request system
- Fixes issues with scanning
- Fixes issues with building
- Adds a resetAll command.

Review please

### CODE REVIEW

**Review by marchermans — DISMISSED** [review]

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java [hunk 1]
```diff
@@ -144,7 +144,7 @@ public static void assignRequest(final IStandardRequestManager manager, final IR
                                         .filter(r -> r.getRequestType().isSupertypeOf(request.getRequestType()))
                                         .filter(r -> !failedResolvers.contains(r.getRequesterId()))
                                         .sorted(Comparator.comparing(r -> -1 * r.getPriority()))
-                                        .collect(Collectors.toSet());
+                                        .collect(Collectors.toCollection(LinkedHashSet::new));
 
             for (final IRequestResolver<?> resolver : resolversForRequestType)
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java:149]: MAJOR Reduce the total number of break and continue statements in this loop to use at most one. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS135)

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java [hunk 2]
```diff
@@ -102,16 +105,19 @@ protected int calculateMaxCraftingCount(@NotNull final ItemStack outputStack, @N
...
+                craftingCount = Math.max(ingredient.getMaxStackSize(), ItemStackUtils.getSize(storage.getPrimaryOutput())) / ItemStackUtils.getSize(storage.getPrimaryOutput());
             }
         }
 
         return craftingCount;
     }
 
     @Nullable
-    protected List<IToken<?>> createRequestsForRecipe(@NotNull final IRequestManager manager, @NotNull final AbstractBuildingWorker building, final ItemStack requestStack, @NotNull final IRecipeStorage storage)
+    protected List<IToken<?>> createRequestsForRecipe(
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java:116]: CRITICAL Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)

File: src/main/java/com/minecolonies/coremod/network/messages/TransferRecipeCrafingTeachingMessage.java [hunk 3]
```diff
@@ -63,4 +60,20 @@ public void toBytes(final ByteBuf buf)
...
+    @Override
+    public void messageOnServerThread(final TransferRecipeCrafingTeachingMessage message, final EntityPlayerMP player)
+    {
+        if (player.openContainer instanceof CraftingGUIBuilding)
+        {
+            final CraftingGUIBuilding container = (CraftingGUIBuilding) player.openContainer;
+
+            container.handleSlotClick(container.getSlot(1), message.itemStacks.containsKey(0) ? message.itemStacks.get(0) : ItemStackUtils.EMPTY);
+            container.handleSlotClick(container.getSlot(2), message.itemStacks.containsKey(1) ? message.itemStacks.get(1) : ItemStackUtils.EMPTY);
+            container.handleSlotClick(container.getSlot(3), message.itemStacks.containsKey(3) ? message.itemStacks.get(3) : ItemStackUtils.EMPTY);
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/network/messages/TransferRecipeCrafingTeachingMessage.java:73]: MAJOR Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
MAJOR Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
MAJOR Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/network/messages/TransferRecipeCrafingTeachingMessage.java [hunk 4]
```diff
@@ -63,4 +60,20 @@ public void toBytes(final ByteBuf buf)
...
+            container.handleSlotClick(container.getSlot(4), message.itemStacks.containsKey(4) ? message.itemStacks.get(4) : ItemStackUtils.EMPTY);
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/network/messages/TransferRecipeCrafingTeachingMessage.java:74]: MAJOR Assign this magic number 4 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
MAJOR Assign this magic number 4 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
MAJOR Assign this magic number 4 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowGuiCrafting.java [hunk 5]
```diff
@@ -4,6 +4,7 @@
 import com.minecolonies.api.util.LanguageHandler;
 import com.minecolonies.api.util.constant.Constants;
 import com.minecolonies.coremod.MineColonies;
+import com.minecolonies.coremod.colony.buildings.AbstractBuildingWorker;
 import com.minecolonies.coremod.colony.buildings.views.AbstractBuildingView;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowGuiCrafting.java]: MINOR Remove this unused import 'com.minecolonies.coremod.colony.buildings.views.AbstractBuildingView'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/compatibility/jei/transer/PrivateCraftingTeachingTransferHandler.java [hunk 6]
```diff
@@ -39,26 +42,34 @@
...
         // indexes that do not fit into the player crafting grid
-        Set<Integer> badIndexes = ImmutableSet.of(2, 5, 6, 7, 8);
+        final Set<Integer> badIndexes = ImmutableSet.of(2, 5, 6, 7, 8);
 
         // compact the crafting grid into a 2x2 area
-        Map<Integer, ItemStack> guiIngredients = new HashMap<>();
+        final Map<Integer, ItemStack> guiIngredients = new HashMap<>();
         guiIngredients.put(0, ItemStackUtils.EMPTY);
         guiIngredients.put(1, ItemStackUtils.EMPTY);
         guiIngredients.put(3, ItemStackUtils.EMPTY);
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/compatibility/jei/transer/PrivateCraftingTeachingTransferHandler.java:60]: MAJOR Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/compatibility/jei/transer/PrivateCraftingTeachingTransferHandler.java [hunk 7]
```diff
@@ -39,26 +42,34 @@
...
         guiIngredients.put(4, ItemStackUtils.EMPTY);
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/compatibility/jei/transer/PrivateCraftingTeachingTransferHandler.java:61]: MAJOR Assign this magic number 4 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 85 issues
* ![CRITICAL][CRITICAL] 9 critical
* ![MAJOR][MAJOR] 69 major
* ![MINOR][MINOR] 7 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuildingView.java#L303](https://github.com/Minecolonies/minecolonies/blob/e940874172a78705cfe86e64e731bc4ce294964b/src/main/java/com/minecolonies/coremod/colony/buildings/views/AbstractBuildingView.java#L303): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuildingView.java#L343](https://github.com/Minecolonies/minecolonies/blob/e940874172a78705cfe86e64e731bc4ce294964b/src/main/java/com/minecolonies/coremod/colony/buildings/views/AbstractBuildingView.java#L343): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [RequestHandler.java#L87](https://github.com/Minecolonies/minecolonies/blob/e940874172a78705cfe86e64e731bc4ce294964b/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L87): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [RequestHandler.java#L89](https://github.com/Minecolonies/minecolonies/blob/e940874172a78705cfe86e64e731bc4ce294964b/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L89): Add a default case to this switch. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
1. ![CRITICAL][CRITICAL] [RequestHandler.java#L115](https://github.com/Minecolonies/minecolonies/blob/e940874172a78705cfe86e64e731bc4ce294964b/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L115): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [RequestHandler.java#L220](https://github.com/Minecolonies/minecolonies/blob/e940874172a78705cfe86e64e731bc4ce294964b/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L220): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractCraftingRequestResolver.java#L77](https://github.com/Minecolonies/minecolonies/blob/e940874172a78705cfe86e64e731bc4ce294964b/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java#L77): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractCraftingRequestResolver.java#L134](https://github.com/Minecolonies/minecolonies/blob/e940874172a78705cfe86e64e731bc4ce294964b/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java#L134): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/e940874172a78705cfe86e64e731bc4ce294964b/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/e940874172a78705cfe86e64e731bc4ce294964b/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/client/gui/WindowGuiCrafting.java [hunk 8]
```diff
@@ -4,7 +4,7 @@
 import com.minecolonies.api.util.LanguageHandler;
 import com.minecolonies.api.util.constant.Constants;
 import com.minecolonies.coremod.MineColonies;
-import com.minecolonies.coremod.colony.buildings.views.AbstractBuildingView;
+import com.minecolonies.coremod.colony.buildings.AbstractBuildingWorker;
 import com.minecolonies.coremod.inventory.CraftingGUIBuilding;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowGuiCrafting.java]: MINOR Remove this unused import 'com.minecolonies.coremod.colony.buildings.views.AbstractBuildingView'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 86 issues
* ![CRITICAL][CRITICAL] 9 critical
* ![MAJOR][MAJOR] 69 major
* ![MINOR][MINOR] 8 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuildingView.java#L303](https://github.com/Minecolonies/minecolonies/blob/162864bd4f4625e5c27b5af90befd0771669fdad/src/main/java/com/minecolonies/coremod/colony/buildings/views/AbstractBuildingView.java#L303): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuildingView.java#L343](https://github.com/Minecolonies/minecolonies/blob/162864bd4f4625e5c27b5af90befd0771669fdad/src/main/java/com/minecolonies/coremod/colony/buildings/views/AbstractBuildingView.java#L343): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [RequestHandler.java#L87](https://github.com/Minecolonies/minecolonies/blob/162864bd4f4625e5c27b5af90befd0771669fdad/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L87): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [RequestHandler.java#L89](https://github.com/Minecolonies/minecolonies/blob/162864bd4f4625e5c27b5af90befd0771669fdad/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L89): Add a default case to this switch. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
1. ![CRITICAL][CRITICAL] [RequestHandler.java#L115](https://github.com/Minecolonies/minecolonies/blob/162864bd4f4625e5c27b5af90befd0771669fdad/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L115): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [RequestHandler.java#L220](https://github.com/Minecolonies/minecolonies/blob/162864bd4f4625e5c27b5af90befd0771669fdad/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L220): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractCraftingRequestResolver.java#L77](https://github.com/Minecolonies/minecolonies/blob/162864bd4f4625e5c27b5af90befd0771669fdad/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java#L77): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractCraftingRequestResolver.java#L134](https://github.com/Minecolonies/minecolonies/blob/162864bd4f4625e5c27b5af90befd0771669fdad/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java#L134): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/162864bd4f4625e5c27b5af90befd0771669fdad/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/162864bd4f4625e5c27b5af90befd0771669fdad/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 85 issues
* ![CRITICAL][CRITICAL] 9 critical
* ![MAJOR][MAJOR] 69 major
* ![MINOR][MINOR] 7 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuildingView.java#L303](https://github.com/Minecolonies/minecolonies/blob/fd24c8eae2f86ef4e3b4f440de0d6fc95b40a530/src/main/java/com/minecolonies/coremod/colony/buildings/views/AbstractBuildingView.java#L303): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuildingView.java#L343](https://github.com/Minecolonies/minecolonies/blob/fd24c8eae2f86ef4e3b4f440de0d6fc95b40a530/src/main/java/com/minecolonies/coremod/colony/buildings/views/AbstractBuildingView.java#L343): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [RequestHandler.java#L87](https://github.com/Minecolonies/minecolonies/blob/fd24c8eae2f86ef4e3b4f440de0d6fc95b40a530/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L87): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [RequestHandler.java#L89](https://github.com/Minecolonies/minecolonies/blob/fd24c8eae2f86ef4e3b4f440de0d6fc95b40a530/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L89): Add a default case to this switch. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
1. ![CRITICAL][CRITICAL] [RequestHandler.java#L115](https://github.com/Minecolonies/minecolonies/blob/fd24c8eae2f86ef4e3b4f440de0d6fc95b40a530/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L115): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [RequestHandler.java#L220](https://github.com/Minecolonies/minecolonies/blob/fd24c8eae2f86ef4e3b4f440de0d6fc95b40a530/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L220): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractCraftingRequestResolver.java#L77](https://github.com/Minecolonies/minecolonies/blob/fd24c8eae2f86ef4e3b4f440de0d6fc95b40a530/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java#L77): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractCraftingRequestResolver.java#L134](https://github.com/Minecolonies/minecolonies/blob/fd24c8eae2f86ef4e3b4f440de0d6fc95b40a530/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java#L134): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/fd24c8eae2f86ef4e3b4f440de0d6fc95b40a530/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/fd24c8eae2f86ef4e3b4f440de0d6fc95b40a530/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**